packages/grafana-ui/src/components/Menu/MenuItem.tsx

```diff
@@ -24,9 +24,20 @@ export interface MenuItemProps {
   /** Active */
   active?: boolean;
 }
+/** This interface was created because we are utilizing two different events in the onClick props*/
```

Comment:
Not sure. understand why here

Reply:
So the onClick handler for ContextMenu is coming from two different sources: menuItem data object and onclose (ContextMenu Props).
We need a way to compose these two actions and that's why we had this conditional logic in the contextMenu component:
```
onClick={(e: React.MouseEvent<HTMLElement>) => {
                  // We can have both url and onClick and we want to allow user to open the link in new tab/window
                  const isSpecialKeyPressed = e.ctrlKey || e.metaKey || e.shiftKey;
                  if (isSpecialKeyPressed && item.url) {
                    return;
                  }

                  if (item.onClick) {
                    e.preventDefault();
                    item.onClick(e);
                  }
```

Now, the challenge is that onclick prop was set to type SynthenticEvent which doesn't correspond with the MouseEvent type we're using here, thus, it requires we create a new interface just to allow onclick accept MouseEvent actions. @torkelo